pkg/apis/core/v1beta1/types_seed.go

```diff
@@ -270,8 +270,23 @@ type SeedSettings struct {
 
 // SeedSettingExcessCapacityReservation controls the excess capacity reservation for shoot control planes in the seed.
 type SeedSettingExcessCapacityReservation struct {
-	// Enabled controls whether the excess capacity reservation should be enabled.
+	// Enabled controls whether the default excess capacity reservation should be enabled.
+	// Deprecated: This field is deprecated and will be removed in a future version of Gardener. Please use `Configs` instead.
 	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
+	// Configs configures excess capacity reservation deployments for shoot control planes in the seed.
+	Configs []SeedSettingExcessCapacityReservationConfig `json:"configs,omitempty" protobuf:"bytes,2,rep,name=configs"`
```

Comment:
From https://github.com/gardener/gardener/blob/master/docs/development/changing-the-api.md#changing-the-api-1:
> Make sure new fields are being added as "optional" fields, i.e., they are of pointer types, they have the `// +optional` comment, and they have the `omitempty` JSON tag.

I guess here the field should have the // +optional tag. Otherwise it would be a breaking change, right?

Reply:
👍 I'll add the comment
tbh, I don't know exactly what `// +optional` is doing. I'll test that. The `omitempty` tag alone makes it non breaking.